_includes/aiml-sidebar.html

```diff
@@ -0,0 +1,129 @@
+        let desktop_sidebar = document.getElementById("desktop-sidebar");
+        let mobile_sidebar = document.getElementById("mobile-sidebar");
+        let window_width = window.innerWidth;
+        let left_margin = desktop_sidebar.offsetLeft;
+
+        left_margin = (window_width / 2.0 - 758) > 26 ? (window_width / 2.0 - 758) : 26;
+        if (window_width < 768) {
+            let hamburger_content = document.getElementById("hamburger-content");
+            if (hamburger_content.style.display != "flex") {
+                document.getElementById("hamburger-content").style.display = "none";
+            }
+            desktop_sidebar.style.display = "none";
+            mobile_sidebar.style.display = "inline-block";
+        } else {
+            desktop_sidebar.style.display = "inline-block";
+            mobile_sidebar.style.display = "none";
+            desktop_sidebar.style.marginLeft = left_margin + "px";
+        }
+    }
+    window.addEventListener('resize', resizeFunctions);
+    toggleSidebarDesktopToMobile();
+
+    document.getElementById("alternate-mobile-icon").addEventListener("click", displayHamburgerMenu);
+    function displayHamburgerMenu() {
+        let hamburger_content = document.getElementById("hamburger-content");
+        let wordsDisplayed = document.getElementById("alternate-mobile-icon");
+        if (hamburger_content.style.display == "flex") {
+            hamburger_content.style.display = "none";
+            wordsDisplayed.innerText = "Lessons";
+        } else {
+            hamburger_content.style.display = "flex";
+            wordsDisplayed.innerText = "Close";
+        }
+
+        // let wordsDisplayed = document.getElementById("alternate-mobile-icon");
+        // if (wordsDisplayed.innerText == "Lessons"){
+        //     wordsDisplayed.innerText = "Close";
+        // } else {
+        //     wordsDisplayed.innerText = "Lessons";
+        // }
+
+        let mobileSidebarButton = document.getElementById("sidebar-btn-container");
+        let window_width = window.innerWidth;
+        if (mobileSidebarButton.style.justifyContent == "flex-end") {
+            mobileSidebarButton.style.justifyContent = "flex-start";
+            if (mobileSidebarButton.style.marginTop != "10px"){
+                mobileSidebarButton.style.marginTop = "10px";
+            }
+        } else {
+            mobileSidebarButton.style.justifyContent = "flex-end";
+            if (window_width <= 386){
+                mobileSidebarButton.style.marginTop = "60px";
+            }
+        }
+    }
+    function adjustMobileButtonHeight() {
+        let mobileSidebarButton = document.getElementById("sidebar-btn-container");
+        let window_width = window.innerWidth;
+        if (window_width <= 386 && mobileSidebarButton.style.justifyContent == "flex-end") {
+            mobileSidebarButton.style.marginTop = "60px";
+        } else {
+            mobileSidebarButton.style.marginTop = "10px";
+        }

```

Comment:
This can be entirely encapsulated in a CSS class, instead of a JS function - media queries are your friend 😊

Reply:
How do i check if it has also been clicked though? Also is there something like :hover but for clicks instead?

Reply:
Here I meant more that when an element is clicked, you can toggle a class - instead of checking the existing styling, screen size, margin, etc.

Reply:
like there would be two separate classes and there would be an JS onclick function that toggles them? Cuz i tried doing the checkbox thing that you did with the navbar but i couldn't get that to work.

Reply:
Yes, two separate classes and a toggle! Using JS for interactivity is totally okay, in fact it's probably recommended!